07-algorithms/algorithms-divide-conquer.md

```diff
@@ -0,0 +1,388 @@
+_Fig. Looking for the position of the value 5 in a sorted array using binary search._
+
+<br />
+
+<details>
+
+<summary>We can view a recursive implementation of binary search by expanding this section.</summary>
+
+```python
+def recursive_binary_search(array, to_find, low=0, high=None):
+    if high is None:
+        high = len(array)
+
+    if high <= low:
+        return None
+
+    mid = (high + low) // 2
+
+    if array[mid] == to_find:
+        return mid
+    elif array[mid] > to_find:
+        high = mid
+    else:
+        low = mid + 1
+
+    return recursive_binary_search(array, to_find, low, high)
+```
+
+</details>
+
+## Check for Understanding
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: 645bad5d-af5f-42a9-a7e4-c70412972ab1
+* title: Time Complexity of Quicksort
+
+##### !question
+
+What is the *Worst-Case* time complexity for Quicksort?
+
+##### !end-question
+
+##### !options
+
+a| O(NlogN)
+b| O(N)
+c| O(N<sup>2</sup>)
+
+##### !end-options
+
+##### !answer
+
+c|
+
+##### !end-answer
+
+
+##### !explanation
+
+The worst-case time complexity for Quicksort is O(N<sup>2</sup>) because picking the worst pivot (e.g. the smallest or biggest remaining element) for each step will result in one array of size n-1 as opposed to two arrays that are approximately n/2 in length.
+
+##### !end-explanation 
+
+### !end-challenge
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: checkbox
+* id: 25522151-56d9-4495-9691-da53700f896e
+* title: Divide and Conquer Algorithms
+
+##### !question
+
+Select all of the following which would be considered Divide/Decrease and Conquer algorithms.
+
+##### !end-question
+
+##### !options
+
+a| Mergesort
+b| Dijkstra's Algorithm
+c| Kadane's Algorithm
+d| Quicksort
+e| Binary Search
+
+##### !end-options
+
+##### !answer
+
+a|
+d|
+e|
+
+##### !end-answer
+
+#### !explanation
+
+Dijkstra's Algorithm is not considered a divide and conquer algorithm because at no point are we breaking the problem into subproblems of the same type and recursively solving the subproblems.
+<br>
+
+Similarly, Kadane's Algorithm is not considered a divide and conquer because we are not breaking the problem into subproblems, but rather using previously calculated results to calculate the overall solution.
```

Comment:
Again, this lesson is actually going to go first, so consider moving this to the DP algorithm as a callback to divide and conquer.

Sequence will be:
- Divide and Conquer
- Greedy
- DP
- Sliding Window

Open to suggestions if you think this should go after DP or we should present things in a different order.